adapter/internal/notifier/deployment_notifier.go

```diff
@@ -49,26 +50,34 @@ func SendRevisionUpdate(deployedRevisionList []*DeployedAPIRevision) {
 	}
 
 	jsonValue, _ := json.Marshal(deployedRevisionList)
-	req, _ := http.NewRequest("POST", revisionEP, bytes.NewBuffer(jsonValue))
+
 	// Setting authorization header
 	basicAuth := "Basic " + auth.GetBasicAuth(cpConfigs.Username, cpConfigs.Password)
-	req.Header.Set("Authorization", basicAuth)
-	req.Header.Set("Content-Type", "application/json")
+
+	logger.LoggerNotifier.Debugf("Revision deployed message sending to Control plane: %v", string(jsonValue))
 
 	// Adding 3 retries for revision update sending
 	retries := 0
 	for retries < 3 {
 		retries++
+
+		req, _ := http.NewRequest("PATCH", revisionEP, bytes.NewBuffer(jsonValue))
+		req.Header.Set("Authorization", basicAuth)
+		req.Header.Set("Content-Type", "application/json")
 		resp, err := tlsutils.InvokeControlPlane(req, cpConfigs.SkipSSLVerification)
+
+		success := true
```

Comment:
Can't we initially set success as false, and then set to true inside the loop

Reply:
We can. I suppose it would add an additional condition checks like follows. 
```if err != nil && resp != nil && resp.StatusCode == http.StatusOK {success=true}```
Since we have already done the negation of the above condition for logging purposes, I guess keeping it as it is would be convenient.